lewis/adapters/epics.py

```diff
@@ -95,6 +94,76 @@ def __init__(self, target_property, poll_interval=1.0, read_only=False,
         self.config = kwargs
 
 
+class BoundPV(object):
+    """
+    Class to represent PVs that are bound to an adapter
+
+    This class is very similar to :class:`~lewis.adapters.stream.Func`, in that
+    it is the result of a binding operation between a user-specified :class:`PV`-object
+    and a Device and/or Adapter object. Also, it should rarely be used directly. objects
+    are generated automatically by :class:`EpicsAdapter`.
+
+    The binding happens by supplying a ``target``-object which has an attribute or a property
+    named according to the property-name stored in the PV-object, and a ``meta_target``-object
+    which has an attribute named according to the meta_data_property in PV.
+
+    The properties ``read_only``, ``config``,  and ``poll_interval`` simply forward the
+    data of PV, while ``doc`` uses the target object to potentially obtain the property's
+    docstring.
+
+    To get and set the value of the property on the target, the ``value``-property of
+    this class can be used, to get the meta data dict, there's a ``meta``-property.
+
+    :param pv: PV object to bind to target and meta_target.
+    :param target: Object that has an attribute named pv.property.
+    :param meta_target: Object that has an attribute named pv.meta_data_property.
+    """
+
+    def __init__(self, pv, target, meta_target=None):
+        self._meta_target = meta_target
+        self._target = target
+        self._pv = pv
+
+    @property
+    def value(self):
+        """Value of the bound property on the target."""
+        return getattr(self._target, self._pv.property)
+
+    @value.setter
+    def value(self, new_value):
+        if not self.read_only:
+            setattr(self._target, self._pv.property, new_value)
+
+    @property
+    def meta(self):
+        """Value of the bound meta-property on the target."""
+        if not self._pv.meta_data_property or not self._meta_target:
+            return {}
+
+        return getattr(self._meta_target, self._pv.meta_data_property)
+
+    @property
+    def read_only(self):
+        """True if the PV is read-only."""
+        return self._pv.read_only
+
+    @property
+    def config(self):
+        """Config dict passed on to pcaspy-machinery."""
+        return self._pv.config
+
+    @property
+    def poll_interval(self):
+        """Interval at which to update PV in pcaspy."""
+        return self._pv.poll_interval
```

Comment:
Yes we should keep an eye on that. It might also happen that `BoundPV` and `PV` diverge, but it's a bit hard to say.